src/agent/src/confidential_data_hub/mod.rs

```diff
@@ -137,6 +172,58 @@ pub async fn unseal_env(env: &str) -> Result<String> {
     Ok((*env.to_owned()).to_string())
 }
 
+/// pull_image is used for call confidential data hub to pull image in the guest.
+/// # Parameters
+/// - `image`: Image name (exp: quay.io/prometheus/busybox:latest)
+/// - `cid`: Container id
+/// - `image_metadata`: Annotations about the image (exp: "containerd.io/snapshot/cri.layer-digest": "sha256:24fb2886d6f6c5d16481dd7608b47e78a8e92a13d6e64d87d57cb16d5f766d63")
```

Comment:
This comment confused me: afaict the only annotation used is the one identifying a sandbox container. Could we just parse that one further up in the call stack? That would also allow implementing pause unpacking and image pulling into separate functions, instead of the `if is_sandbox` switch here.

Reply:
It's also a copy-pasted old function. FWIW, this API wraps all image pull logics inside the function and other modules calls this function thus looks not too bad to me. But we could make it better. Probably we could do this in another PR?

Reply:
Yup. This is also some copy-paste work, but I'd like to do refac in the PR too